fastapi/routing.py

```diff
@@ -968,6 +975,10 @@ def add_api_websocket_route(
             dependencies=current_dependencies,
             dependency_overrides_provider=self.dependency_overrides_provider,
         )
+        hash_val = get_path_hash_val(route.path)
+        if hash_val in self.added_routes:
+            raise RouteAlreadyExistsError(route.name)
```

Comment:
We can do it without additional storage, something like this:
```python3
is_route_exist = [exist_route 
                  for exist_route in self.routes
                  if route.path == self.path and route.methods == self.methods]
if is_route_exist:
    raise RouteAlreadyExistsError(route.name)

Reply:
Not sure if this would work. `self.path` is not in the context of this function.